gtk4/src/subclass/cell_layout.rs

```diff
@@ -0,0 +1,489 @@
+    fn parent_set_cell_data_func<R: IsA<CellRenderer>>(
+        &self,
+        cell_layout: &Self::Type,
+        cell: &R,
+        callback: Option<&CellLayoutDataCallback>,
+    );
+    fn parent_reorder<R: IsA<CellRenderer>>(
+        &self,
+        cell_layout: &Self::Type,
+        cell: &R,
+        position: i32,
+    );
+    fn parent_clear(&self, cell_layout: &Self::Type);
+    fn parent_pack_start<R: IsA<CellRenderer>>(
+        &self,
+        cell_layout: &Self::Type,
+        cell: &R,
+        expand: bool,
+    );
+    fn parent_pack_end<R: IsA<CellRenderer>>(
+        &self,
+        cell_layout: &Self::Type,
+        cell: &R,
+        expand: bool,
+    );
+    fn parent_area(&self, cell_layout: &Self::Type) -> Option<CellArea>;
+}
+
+impl<O: CellLayoutImpl> CellLayoutImplExt for O {
+    fn parent_add_attribute<R: IsA<CellRenderer>>(
+        &self,
+        cell_layout: &Self::Type,
+        cell: &R,
+        attribute: &str,
+        column: i32,
+    ) {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface = type_data.as_ref().parent_interface::<CellLayout>()
+                as *const ffi::GtkCellLayoutIface;
+
+            if let Some(f) = (*parent_iface).add_attribute {
+                f(
+                    cell_layout.unsafe_cast_ref::<CellLayout>().to_glib_none().0,
+                    cell.as_ref().to_glib_none().0,
+                    attribute.to_glib_none().0,
+                    column,
+                );
+            }
+        }
+    }
+
+    fn parent_clear_attributes<R: IsA<CellRenderer>>(&self, cell_layout: &Self::Type, cell: &R) {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface = type_data.as_ref().parent_interface::<CellLayout>()
+                as *const ffi::GtkCellLayoutIface;
+
+            if let Some(f) = (*parent_iface).clear_attributes {
+                f(
+                    cell_layout.unsafe_cast_ref::<CellLayout>().to_glib_none().0,
+                    cell.as_ref().to_glib_none().0,
+                );
+            }
+        }
+    }
+
+    fn parent_cells(&self, cell_layout: &Self::Type) -> Vec<CellRenderer> {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface = type_data.as_ref().parent_interface::<CellLayout>()
+                as *const ffi::GtkCellLayoutIface;
+
+            let f = (*parent_iface)
+                .get_cells
+                .as_ref()
+                .expect("no parent \"get_cells\" implementation");
+
+            let cells = f(cell_layout.unsafe_cast_ref::<CellLayout>().to_glib_none().0);
+            FromGlibPtrArrayContainerAsVec::from_glib_container_as_vec(cells)
+        }
+    }
+
+    fn parent_set_cell_data_func<R: IsA<CellRenderer>>(
+        &self,
+        cell_layout: &Self::Type,
+        cell: &R,
+        callback: Option<&CellLayoutDataCallback>,
```

Comment:
```suggestion
        callback: Option<CellLayoutDataCallback>,
```

And here also to prevent double frees.